File: JavaFactorial/src/de/luschny/math/factorial/FactorialProductNaive.java
// Copyright (C) 2004-2009 Peter Luschny, MIT License applies.
// See http://en.wikipedia.org/wiki/MIT_License
// Visit http://www.luschny.de/math/factorial/FastFactorialFunctions.htm
// Comments mail to: peter(at)luschny.de
package de.luschny.math.factorial;

import de.luschny.math.arithmetic.Xint;

public class FactorialProductNaive implements IFactorialFunction {

    public FactorialProductNaive() {
    }

    @Override
    public String getName() {
        return "ProductNaive        ";
    }

    @Override
    public Xint factorial(int n) {
        if (n < 0) {
            throw new ArithmeticException("Factorial: n has to be >= 0, but was " + n);
        }

        Xint nFact = Xint.ONE;

        for (int i = 2; i <= n; i++) {
            nFact = nFact.multiply(i);
        }

        return nFact;
    }
} // endOfFactorialProductNaive

File: SilverFactorial64/Sharith/Factorial/FactorialProductNaive.cs
// -------- ToujoursEnBeta
// Author & Copyright : Peter Luschny
// License: LGPL version 3.0 or (at your option)
// Creative Commons Attribution-ShareAlike 3.0
// Comments mail to: peter(at)luschny.de
// Created: 2010-03-01

namespace Sharith.Factorial 
{
    using XInt = Arithmetic.XInt;

    public class ProductNaive : IFactorialFunction  
    {
        public string Name => "ProductNaive        ";

        public XInt Factorial(int n)
        {
            if (n < 0)
            {
                throw new System.ArgumentOutOfRangeException(
                    this.Name + ": " + nameof(n) + " >= 0 required, but was " + n);
            }

            var nFact = XInt.One;

            for (var i = 2; i <= n; i++)
            {
                nFact *= i;
            }
            return nFact;
        }
    }
} // endOfProductNaive
